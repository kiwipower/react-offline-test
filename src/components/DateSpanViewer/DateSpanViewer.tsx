import React from 'react'
import { format, parseISO } from 'date-fns';

interface DateSpanViewerProps {
    from: string;
    to: string;
}

export default function DateSpanViewer(props: DateSpanViewerProps) {

    if (!props.from) {
        throw "from props is empty";
    }

    if (!props.to) {
        throw "to props is empty";
    }

    const fromParsed =  parseISO(props.from);
    const toParsed =  parseISO(props.to);

    return (
        <div className="card-body">
            <div className="row justify-content-center">
                <div className="col-sm-2 date-container__item">
                    {format(fromParsed, 'dd-MM-yyy')}
                    <span>{format(fromParsed, 'HH:mm')}</span>
                </div>
                <div className="col-sm-2 date-container__item">
                    {format(toParsed, 'dd-MM-yyy')}
                    <span>{format(toParsed, 'HH:mm')}</span>
                </div>
            </div>
        </div>
    )
}
