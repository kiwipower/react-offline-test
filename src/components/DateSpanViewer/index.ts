import  DateSpanViewer from "./DateSpanViewer";

export default DateSpanViewer;