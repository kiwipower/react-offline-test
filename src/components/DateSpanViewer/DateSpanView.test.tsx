import React from 'react';
import { shallow } from 'enzyme';

import DateSpanViewer from './DateSpanViewer';

describe('<DateSpanViewer />', () => {

    const renderComponent = () => (<DateSpanViewer from="2019-08-12T12:30Z" to="2019-08-12T13:00Z" />);

    it('renders without error', async () => {
        const component = shallow(renderComponent());
        expect(component).toMatchSnapshot();
    });

    it('date format working correctly', () => {
        const component = shallow(renderComponent());
        const fromHour = component
            .find('.date-container__item').at(0)
            .find('span').text()
        expect(fromHour).toBe('13:30'); //BTZ
    })

});