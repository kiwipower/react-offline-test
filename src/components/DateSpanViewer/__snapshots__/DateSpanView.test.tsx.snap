// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateSpanViewer /> renders without error 1`] = `
<div
  className="card-body"
>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-sm-2 date-container__item"
    >
      12-08-2019
      <span>
        13:30
      </span>
    </div>
    <div
      className="col-sm-2 date-container__item"
    >
      12-08-2019
      <span>
        14:00
      </span>
    </div>
  </div>
</div>
`;
